File: ToDoApp/ToDoApp.Web/Modules/ServerTypes/TaskDB/TaskColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { Status } from "../Modules/TaskDB.Task.Status";
import { TaskRow } from "./TaskRow";

export interface TaskColumns {
    TaskId: Column<TaskRow>;
    Title: Column<TaskRow>;
    Description: Column<TaskRow>;
    Status: Column<TaskRow>;
    Priority: Column<TaskRow>;
    DateCreated: Column<TaskRow>;
    DueDate: Column<TaskRow>;
}

export class TaskColumns extends ColumnsBase<TaskRow> {
    static readonly columnsKey = 'TaskDB.Task';
    static readonly Fields = fieldsProxy<TaskColumns>();
}

[Status]; // referenced types

File: ToDoApp/ToDoApp.Web/Modules/TaskDB/Task/TaskColumns.cs
using Serenity.ComponentModel;
using System.ComponentModel;
using ToDoApp.Modules.TaskDB.Task;

namespace ToDoApp.TaskDB.Columns;


[ColumnsScript("TaskDB.Task")]
[BasedOnRow(typeof(TaskRow), CheckNames = true)]
public class TaskColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int TaskId { get; set; }
    [EditLink]
    public string Title { get; set; }
    public string Description { get; set; }
    [Width(80), AlignCenter]
    public Status Status { get; set; }
    public Priority Priority { get; set; }
    [DisplayName("Date Created")]
    public DateTime? DateCreated { get; set; }

    [DisplayName("Due Date")]
    public DateTime? DueDate { get; set; }


}